lib/rules/object-curly-newline.js

```diff
@@ -206,6 +219,10 @@ module.exports = {
                         node,
                         loc: closeBrace.loc.start,
                         fix(fixer) {
+                            if (hasComments) {
```

Comment:
For this check, I think it would matter whether the *last* token of the object is a comment, not the first.